src/pilot/config_idrac.py

```diff
@@ -204,6 +207,199 @@ def configure_bios_nics_boot_settings(drac_client, ip_service_tag, pxe_nic_id):
+    """
+    return set_nic_setting(nic_id, 'LegacyBootProto', value, drac_client)
+
+
+def set_nic_legacy_boot_protocol_none(nic_id, drac_client):
+    """Set the legacy, non-UEFI, boot protocol of a NIC to NONE.
+
+    If successful, the pending value of the NIC's legacy boot
+    protocol attribute is set. For the new value to be applied, a
+    configuration job must be created and the node must be rebooted.
+
+    :param nic_id: id of the network interface controller (NIC)
+    :param drac_client: drac_client from python-dracclient
+    :returns: dictionary containing a 'commit_required' key with a
+              boolean value indicating whether a configuration job
+              must be created for the new legacy boot protocol
+              setting to be applied
+    :raises: WSManRequestFailure on request failures
+    :raises: WSManInvalidResponse when receiving invalid response
+    :raises: DRACOperationFailed on error reported back by the iDRAC
+             interface
+    """
+    return set_nic_legacy_boot_protocol(nic_id, 'NONE', drac_client)
+
+
+def set_nic_legacy_boot_protocol_pxe(nic_id, drac_client):
+    """Set the legacy, non-UEFI, boot protocol of a NIC to PXE.
+
+    If successful, the pending value of the NIC's legacy boot
+    protocol attribute is set. For the new value to be applied, a
+    configuration job must be created and the node must be rebooted.
+
+    :param nic_id: id of the network interface controller (NIC)
+    :param drac_client: drac_client from python-dracclient
+    :returns: dictionary containing a 'commit_required' key with a
+              boolean value indicating whether a configuration job
+              must be created for the new legacy boot protocol
+              setting to be applied
+    :raises: WSManRequestFailure on request failures
+    :raises: WSManInvalidResponse when receiving invalid response
+    :raises: DRACOperationFailed on error reported back by the iDRAC
+             interface
+    """
+    return set_nic_legacy_boot_protocol(nic_id, 'PXE', drac_client)
+
+
+def _list_nic_settings(selection_expression):
+    result = {}
+    configurable_attributes = [
+        (uris.DCIM_NICEnumeration,
+         'DCIM_NICEnumeration',
+         nic.NICEnumerationAttribute),
+        (uris.DCIM_NICString,
+         'DCIM_NICString',
+         nic.NICStringAttribute),
+        (uris.DCIM_NICInteger,
+         'DCIM_NICInteger',
+         nic.NICIntegerAttribute)]
+
+    for (resource, class_name, attr_cls) in configurable_attributes:
+        attribs = _get_nic_config(resource, class_name,
+                                  selection_expression, attr_cls)
+
+        if not set(result).isdisjoint(set(attribs)):
+            raise exceptions.DRACOperationFailed(
+                drac_messages=('Colliding attributes %r' % (
+                    set(result) & set(attribs))))
+
+        result.update(attribs)
+
+    return result
+
+
+def get_nic_setting(nic_id, attribute_name):
```

Comment:
It's ok to have this method, but it needs to be changed to call the upstream list_nic_settings(), and then find the given NIC setting in the resulting list.